src/Symfony/Component/PasswordHasher/Hasher/UserPasswordHasher.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\PasswordHasher\Hasher;
+
+use Symfony\Component\Security\Core\User\UserInterface;
+
+/**
+ * A generic password hasher.
```

Comment:
```suggestion
 * Hashes passwords based on the user and the HasherFactory.
```

"generic" doesn't say much :)